src/http/client.cc

```diff
@@ -153,13 +153,26 @@ client::client(std::unique_ptr<connection_factory> f)
 }
 
 future<client::connection_ptr> client::get_connection() {
+    if (!_pool.empty()) {
+        connection_ptr con = _pool.front().shared_from_this();
+        _pool.pop_front();
+        con->_pooled = {};
+        return make_ready_future<connection_ptr>(con);
+    }
```

Comment:
What about this:

```
struct connection_list_entry {
    hook_t hook;
    connection_ptr ptr;
    
    explicit(compaction_ptr ptr_) : ptr(std::move(ptr_)) {}
}
```

so we can decouple ownership of X from X itself.

Reply:
It's then double allocation per connection. But what's the gain?